catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/resource/download/ResourceDownloadEndpoint.java

```diff
@@ -47,17 +63,76 @@
 
     private final ReliableResourceDownloadManager downloadManager;
 
+    private final ObjectMapper objectMapper;
+
+    static class ResourceDownloadResponse {
+        private String downloadIdentifier;
+
+        private ResourceDownloadResponse(String downloadIdentifier) {
+            this.downloadIdentifier = downloadIdentifier;
+        }
+
+        public String getDownloadIdentifier() {
+            return downloadIdentifier;
+        }
+    }
+
     public ResourceDownloadEndpoint(CatalogFramework catalogFramework,
-            ReliableResourceDownloadManager downloadManager) {
+            ReliableResourceDownloadManager downloadManager, ObjectMapper objectMapper) {
         this.catalogFramework = catalogFramework;
         this.downloadManager = downloadManager;
+        this.objectMapper = objectMapper;
     }
 
+    /**
+     * Gets the list of active product downloads, or starts an asynchronous download of a specific
+     * metacard product to the product cache if called with query parameters.
+     * <p/>
+     * This method supports both, starting new downloads and returning the current list of active
```

Comment:
With the comma after "both", I'm inclined to ask "both" what? If you remove the comma, this reads properly. Or replace the comma with a semi-colon to achieve the same effect.